lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -247,6 +249,9 @@ instance TxWitnessTagFor IcarusKey where
 instance TxWitnessTagFor ByronKey where
     txWitnessTagFor = TxWitnessByronUTxO Byron
 
+instance TxWitnessTagFor SharedKey where
```

Comment:
This instance looks *very* suspicious to me. This should not be needed for Shared wallets since they are signed independently and using a different process. Plus, `TxWitnessTagFor` is used for calculating the size of the resulting transaction; in the case of shared scripts this also needs to acknowledge for the script in the witness which require more changes to the code down the line.

Reply:
removed newTransactionLayer from apiLayer for the time being. Will return to it when txs are to be supported for shared wallets